extensions/notebook/src/book/remoteBook.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as loc from '../common/localizedConstants';
+import * as vscode from 'vscode';
+import * as utils from '../common/utils';
+import { IAsset } from './remoteBookController';
+
+export abstract class RemoteBook {
+	protected _localPath: URL;
+	protected outputChannel: vscode.OutputChannel;
+
+	constructor(public remotePath: URL, protected _asset?: IAsset) {
+		this.remotePath = remotePath;
+		this.outputChannel = vscode.window.createOutputChannel(loc.msgTaskName);
+	}
+
+	public async abstract createLocalCopy(): Promise<void>;
+
+	public setLocalPath(): void {
```

Comment:
We should probably inform the user that the book will be saved to worspaceFolder/userHome.